io/io/src/TFile.cxx

```diff
@@ -318,6 +318,26 @@ TFile::TFile(const char *fname1, Option_t *option, const char *ftitle, Int_t com
    if (!gROOT)
       ::Fatal("TFile::TFile", "ROOT system not initialized");
 
+   auto zombify = [this] {
+      // error in file opening occurred, make this object a zombie
+      if (fGlobalRegistration) {
+         R__LOCKGUARD(gROOTMutex);
+         gROOT->GetListOfClosedObjects()->Add(this);
```

Comment:
```suggestion
         gROOT->GetListOfClosedObjects()->Add(this);
```
The list of Closed Objects has its won locked nowadays (yes, I don't know why it was not removed sooner).

Reply:
this looks completely unrelated, it should probably be done in a separate PR for all occurrences of the pattern